logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueReader.java

```diff
@@ -331,7 +331,7 @@ private void removeSegmentsBefore(Path validSegment) throws IOException {
         final Comparator<Path> fileTimeAndName = ((Comparator<Path>) this::compareByFileTimestamp)
                 .thenComparingInt(DeadLetterQueueUtils::extractSegmentId);
 
-        try (final Stream<Path> segmentFiles = Files.list(queuePath)) {
+        try (final Stream<Path> segmentFiles = DeadLetterQueueWriter.getSegmentPaths(queuePath)) {
```

Comment:
We can remove the `DeadLetterQueueWriter.` from here - we have a static import for this method, and we're using it elsewhere

Reply:
Have aligned this in commit https://github.com/elastic/logstash/pull/14256/commits/fc2a2f9a01399a6d2c0bd09cb0d4313d968affa7